### ISSUE TITLE
Older Firefox versions have trouble reading value from <input> elements

### ISSUE THREAD

**Issue by craigfowler** [csf-dev/CSF.Screenplay.Selenium#26]
This has been detected on Firefox version **51.0.1**.  The test scenario **Reading the value of a form element: Reading the value of an element and converting it to a number detects the expected value.** fails on this browser version, because the response from the remote web driver (when reading the value) is empty string instead of the expected 55.

## Test failure with exception
This causes the following test failure and exception.
```
) Error : CSF.Screenplay.Selenium.Tests.Questions.GetValueTests.GetConvertedValue_returns_expected_value(CSF.Screenplay.Actors.Cast,Value is not created.)
System.FormatException : Input string was not in a correct format.
  at System.Number.StringToNumber (System.String str, System.Globalization.NumberStyles options, System.Number+NumberBuffer& number, System.Globalization.NumberFormatInfo info, System.Boolean parseDecimal) [0x00057] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.Number.ParseInt32 (System.String s, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) [0x00015] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.Int32.Parse (System.String s, System.Globalization.NumberStyles style, System.IFormatProvider provider) [0x0000e] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.Convert.ToInt32 (System.String value, System.IFormatProvider provider) [0x00005] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.String.System.IConvertible.ToInt32 (System.IFormatProvider provider) [0x00000] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x0011a] in <2e7c1c96edae44d496118948ca617c11>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType) [0x0000c] in <2e7c1c96edae44d496118948ca617c11>:0 
  at CSF.Screenplay.Selenium.Queries.ValueQuery`1[T].GetElementData (CSF.Screenplay.Selenium.Models.IWebElementAdapter adapter) [0x00008] in <6b0b0f425236495c829e62ec1426dd18>:0 
  at CSF.Screenplay.Selenium.Queries.Query`1[T].CSF.Screenplay.Selenium.Queries.IQuery<T>.GetElementData (CSF.Screenplay.Selenium.Models.IWebElementAdapter adapter) [0x00001] in <6b0b0f425236495c829e62ec1426dd18>:0 
  at CSF.Screenplay.Selenium.Questions.TargettedQuestion`1[T].GetAnswer (CSF.Screenplay.Actors.IPerformer actor, CSF.Screenplay.Selenium.Abilities.BrowseTheWeb ability, CSF.Screenplay.Selenium.Models.IWebElementAdapter adapter) [0x00001] in <6b0b0f425236495c829e62ec1426dd18>:0 
  at CSF.Screenplay.Selenium.Questions.TargettedQuestion`1[T].PerformAs (CSF.Screenplay.Actors.IPerformer actor) [0x00011] in <6b0b0f425236495c829e62ec1426dd18>:0 
  at CSF.Screenplay.Performables.Performable`1[TResult].CSF.Screenplay.Performables.IPerformable<TResult>.PerformAs (CSF.Screenplay.Actors.IPerformer actor) [0x0000e] in <65c017afcc0a456da82fc37802bd2e37>:0 
  at CSF.Screenplay.Actors.Actor.Perform[TResult] (CSF.Screenplay.Performables.IPerformable`1[TResult] performable) [0x0003d] in <65c017afcc0a456da82fc37802bd2e37>:0 
  at CSF.Screenplay.Actors.Actor.CSF.Screenplay.Actors.IThenActor.ShouldSee[TResult] (CSF.Screenplay.Performables.IQuestion`1[TAnswer] question) [0x00006] in <65c017afcc0a456da82fc37802bd2e37>:0 
  at CSF.Screenplay.Selenium.Tests.Questions.GetValueTests.GetConvertedValue_returns_expected_value (CSF.Screenplay.Actors.ICast cast, System.Lazy`1[T] webBrowserFactory) [0x0004f] in <3f21be990e1646a686656091e3462cf2>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <2e7c1c96edae44d496118948ca617c11>:0 
```

**Comment by craigfowler** [comment]
## Versions affected
Firefox **51.0** and **52.0** are affected.  *Version 53.0 is not affected*.

**Comment by craigfowler** [comment]
This might be fixable by manually firing the change and/or event(s).  In theory that can be done by using JavaScript and the solution documented here:
* https://stackoverflow.com/a/27277924/6221779
* https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/activeElement

**Comment by craigfowler** [comment]
Blocked by https://github.com/csf-dev/WebDriverExtras/issues/32 though, because at present we can't get a version number for Firefox in order to set an appropriate Browser Flag.

**Comment by craigfowler** [comment]
In this ticket, all I want to do is:
* Confirm that the failure is real
* Research the root cause (I suspect it's because the change/blur event never fired)
* Open a milestone 0.3 ticket to create a JavaScript workaround (if applicable)
* Ignore the test

A JavaScript workaround will need to be a 0.3 ticket, because of issue #36.

**Comment by craigfowler** [comment]
Actually, turns out that this is a compatibility issue and that it's nothing to do with events at all.  Firefox 51 and 52 don't support getting the *current* values of `<input>` elements.  They will only read the value attribute and not the current information.  This makes it all-but impossible to support those versions.  See https://github.com/mozilla/geckodriver/issues/186

**Comment by craigfowler** [comment]
I'm going to WONTFIX this and drop support for Firefox 51 and 52.
